### PR TITLE
Switch mails to whitelisting

### PR DESCRIPTION
**Pull request by kortewegdevries**
I might do kmail too, if I can. Will be same format, pushed in this PR.

### CODE REVIEW

File: etc/profile-a-l/evolution.profile [hunk 1]
```diff
@@ -40,7 +72,25 @@ novideo
...
+private-bin evolution
+private-cache
 private-dev
+private-etc alternatives,ca-certificates,crypto-policies,dconf,fonts,gcrypt,gtk-2.0,gtk-3.0,groups,hostname,hosts,mailname,passwd,pki,resolv.conf,selinux,ssl,xdg
 private-tmp
+writable-run-user
 writable-var
+
+dbus-user filter
+dbus-user.own org.gnome.Evolution
```
Comment (rusty-snake) [inline etc/profile-a-l/evolution.profile:88]: The flatpak has also
```
org.gnome.keyring.SystemPrompter=talk
org.gnome.OnlineAccounts=talk
```
I'm not sure if they are needed.

File: etc/profile-a-l/evolution.profile [hunk 2]
```diff
@@ -40,7 +72,25 @@ novideo
...
+dbus-user.talk ca.desrt.dconf
+# Uncomment to have keyring access
+# dbus-user.talk org.freedesktop.secrets
+dbus-user.talk org.freedesktop.Notifications
+dbus-system none
+
+# Comment to use gpg
+read-only ${HOME}/.gnupg
```
Comment (rusty-snake) [inline etc/profile-a-l/evolution.profile:96]: ~/.gnupg is blacklisted by default in your current commit. Why these?

File: etc/profile-a-l/evolution.profile [hunk 3]
```diff
@@ -40,7 +72,25 @@ novideo
 protocol unix,inet,inet6
 seccomp
 shell none
+tracelog
 
+disable-mnt
+# Add "gpg,gpg2,gpg-agent,pinentry-curses,pinentry-emacs,pinentry-fltk,pinentry-gnome3,pinentry-gtk,pinentry-gtk2,pinentry-gtk-2,pinentry-qt,pinentry-qt4,pinentry-tty,pinentry-x2go,pinentry-kwallet" for gpg
+private-bin evolution
```
Comment (rusty-snake) [inline etc/profile-a-l/evolution.profile:79]: Probably breaks others things too?

File: etc/profile-a-l/evolution.profile [hunk 4]
```diff
@@ -40,7 +72,25 @@ novideo
 protocol unix,inet,inet6
 seccomp
 shell none
+tracelog
 
+disable-mnt
```
Comment (rusty-snake) [inline etc/profile-a-l/evolution.profile:77]: Is it common to attach files from USB-Sticks?

**Review by rusty-snake — COMMENTED** [review]
- Whitelisting: +1
- Disabling gpg support: I'm not sure whether this is good. I mean if the most users will need to enable it, it would be better to keep it and the users who don't need it can remove it if they want.

File: etc/profile-a-l/evolution.profile (hunk 1 above)
Comment (kortewegdevries) [reply etc/profile-a-l/evolution.profile:88]: The prompter is for the password box I think, in this regard seahorse needs to be fixed, exporting secret key doesn't work with current dbus-filters, it's probably x.x.x.internalprompter something. I don't know about online accounts...

File: etc/profile-a-l/evolution.profile (hunk 2 above)
Comment (kortewegdevries) [reply etc/profile-a-l/evolution.profile:96]: Copy/paste error...

File: etc/profile-a-l/evolution.profile (hunk 3 above)
Comment (kortewegdevries) [reply etc/profile-a-l/evolution.profile:79]: I haven't seen except browsers,I used plain-text emails. I'll comment bin and remove the gpg line?

File: etc/profile-a-l/evolution.profile (hunk 4 above)
Comment (kortewegdevries) [reply etc/profile-a-l/evolution.profile:77]: Is it? Idk...

File: etc/profile-a-l/evolution.profile (hunk 3 above)
Comment (rusty-snake) [reply etc/profile-a-l/evolution.profile]: I would say add the gpg stuff to the commented private-bin for references if someone wants to use private-bin.

**Comment by kortewegdevries** [comment]
So I will rewrite the geary profile since it was decided not to have a firefox redirect (ref. https://github.com/netblue30/firejail/issues/  1773#issuecomment-364792590) The dbus-filters needs to be added, since I could find one issue regarding dbus on Arch.

**Comment by rusty-snake** [comment]
> is it common to open a link in mail?

yes

**Comment by kortewegdevries** [comment]
So do you want to add profiles.ini,dbus talk,commented private-bin bash,sh,which,firefox to them?

**Comment by rusty-snake** [comment]
If the profile has no private-bin, add the lines from thunderbird.
```
noblacklist ${HOME}/.mozilla
whitelist ${HOME}/.mozilla/firefox/profiles.ini
read-only ${HOME}/.mozilla/firefox/profiles.ini
```
If it has a private-bin add it commented and a note about private-bin that you need to ignore or extend.

**Comment by kortewegdevries** [comment]
If there aren't any corrections, I'm done on both mails: (2+3)...

File: etc/profile-a-l/evolution.profile [hunk 5]
```diff
@@ -71,7 +73,8 @@ shell none
 tracelog
 
 # disable-mnt
-# Add "gpg,gpg2,gpg-agent,pinentry-curses,pinentry-emacs,pinentry-fltk,pinentry-gnome3,pinentry-gtk,pinentry-gtk2,pinentry-gtk-2,pinentry-qt,pinentry-qt4,pinentry-tty,pinentry-x2go,pinentry-kwallet" for gpg 
+# Add "gpg,gpg2,gpg-agent,pinentry-curses,pinentry-emacs,pinentry-fltk,pinentry-gnome3,pinentry-gtk,pinentry-gtk2,pinentry-gtk-2,pinentry-qt,pinentry-qt4,pinentry-tty,pinentry-x2go,pinentry-kwallet" for gpg
+# Add "sh,bash,which,firefox" for hyperlinks
```
Comment (rusty-snake) [inline etc/profile-a-l/evolution.profile:77]: The private-bin for firefox is distro-specific, this works only for Arch.

Comment (kortewegdevries) [reply etc/profile-a-l/evolution.profile:77]: xdg-open or something else? I use this on various profiles on Xubuntu with 0.9.64-2,assuming a firefox is running...

Comment (kortewegdevries) [reply etc/profile-a-l/evolution.profile:77]: Since it's distro dependent, I just remove that line?

**Comment by netblue30** [comment]
@rusty-snake should we pull it in?

**Comment by rusty-snake** [comment]
IDK, I didn't see something against it. Testing it would be the best.

File: etc/profile-a-l/evolution.profile [hunk 6]
```diff
@@ -40,7 +70,27 @@ novideo
 protocol unix,inet,inet6
 seccomp
 shell none
+tracelog
 
+# disable-mnt
+# Add "gpg,gpg2,gpg-agent,pinentry-curses,pinentry-emacs,pinentry-fltk,pinentry-gnome3,pinentry-gtk,pinentry-gtk2,pinentry-gtk-2,pinentry-qt,pinentry-qt4,pinentry-tty,pinentry-x2go,pinentry-kwallet" for gpg
```
Comment (Fred-Barclay) [inline etc/profile-a-l/evolution.profile:76]: If private-bin doesn't support gpg by default (which really I think we should support) then shouldn't we block access to ~/.gnupg ?

Comment (rusty-snake) [reply etc/profile-a-l/evolution.profile:76]: `private-bin` is disabled (see https://github.com/netblue30/firejail/pull/3607#discussion_r479185506), this means gpg is in $PATH.

Comment (Fred-Barclay) [reply etc/profile-a-l/evolution.profile:76]: @rusty-snake 🤦 yep!

**Comment by hyiltiz** [comment]
I can test out kmail if/when it is ready.

**Comment by rusty-snake** [comment]
Should we merge now?

**Comment by netblue30** [comment]
all in, thanks!

**Pull request merged**